### PR TITLE
Add JBrowse2

### PR DESCRIPTION
**Pull request by bgruening**
This PR is adding JBrowse2. It tries to merge https://github.com/galaxyproject/tools-iuc/pull/5695 with @abretaud work from https://github.com/galaxyproject/tools-iuc/pull/3997/.

I tried my best to keep all the features from both branches, but I'm aware its not perfect.
However, I think it's in a solid state and was tested by various people from the VGP and ERGA projects, so I think we should deploy it properly.

It's a super exciting but complicated tool, so I assume we will see many updates over time as we did with JBrowse1.

One major improvement over version 1 is that we have now proper, more complicated tests. We use the `location` attribute in Galaxy tests now and create complicated tracks. Test data is currently [here](https://usegalaxy.eu/static/share/ebp/jbrowse2/test-data/). The resulting zip file is then extensively "inspected". I think we can use this in the future to increase the test coverage.

A few other things in this PR:
  * HiC, Juicebox HiC, PAF, CRAM ... tracks added
  * The requested repeats from https://github.com/galaxyproject/tools-iuc/pull/5695 are included.
  * I added `<required_files>` so hopefully this helps with Pulsar
  * added `creators` with Ross, Helena and Anthony

Thanks to @abretaud, @hexylena  and @fubar2 for working on this.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by hexylena — CHANGES_REQUESTED** [review]
We, appreciate, you trying to resolve this. However, we still feel uncomfortable with this PR, as we've had (internal) discussions around what needs to be part of a jbrowse2 PR.

What you've done here seems to be to incorporate Ross' changes *on top* of the work in https://github.com/galaxyproject/tools-iuc/pull/3997, replacing it in many places.

That code as it was, did not fit with what the genome annotation community feels is important moving forward. I refer you to my review comments in https://github.com/galaxyproject/tools-iuc/pull/5697#pullrequestreview-1836437432, this PR does not address any of those concerns. Namely: it still uses JSON to configure tracks which is fragile and undesirable, when there are solid CLI commands that we can be sure will continue working going forward if they make internal structure changes. Ross reports (unfiled) issues with the CLI tools, we should work to resolve those for the health of the community and use *limited* exceptions where we need things sooner.

by making a separate PR you've taken away from the diff that would be visible, had you made a PR to Anthony's branch in #3997. So it is harder to see what is being added to or changed from that.

If you *really* want the community to accept this, I would request that you:

- [ ] Close this PR and re-open it against [anthony's branch](https://github.com/abretaud/tools-iuc/tree/jbrowse2). This is needed so we can see what changes are being made to that existing extremely solid base of a PR and discuss those on a case by case basis.
- [ ] Revert to using the CLI, rather than JSON blobs that are not as future proof.

We need to review this in more detail,  and I think there are other bits of code smell like the SSL context that give me pause. I've made some small review comments but it is not a full review, I'll save that for when we can see a reduced diff of what has changed against the existing PR.

File: tools/jbrowse2/jbrowse2.xml [hunk 1]
```diff
@@ -0,0 +1,1240 @@
...
+]]></configfile>
+    </configfiles>
+    <inputs>
+        <repeat name="assemblies" min="1" title="Genome reference to provide display coordinates for this set of tracks" help="JBrowse2 can show a set of tracks for each of multiple genome reference fasta files">
+            <conditional name="reference_genome">
+                <param name="genome_type_select" type="select" label="Reference genome source" help="Select a built in, history or remote tabix URI for the reference track">
+                    <option selected="True" value="indexed">Use a Galaxy server built-in genome</option>
+                    <option value="history">Use a genome fasta file from the current history</option>
+                    <option value="uri">Provide a URI (e.g. "https://..." for a genome reference as
+                        tabix bgzip with predictable index file URI</option>
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/jbrowse2/jbrowse2.xml:324]: Afaik, this was requested by users to create JBrowse2 (relocateable) archives that point to public data. In this case it was the GenomeArk, where all assembly data from VGP ends up. I think the argumentation was, the data should not be stored on Galaxy for every, so they need to move. But at the same point, there is a need to add a JBrowse2 instance to the GenomeArk that still works. I hope I recalled the feature correctly.

File: tools/jbrowse2/jbrowse2.py [hunk 2]
```diff
@@ -0,0 +1,1753 @@
...
+                },
+                "sequenceAdapter": genseqad,
+            },
+            "displays": [
+                {
+                    "type": "LinearAlignmentsDisplay",
+                    "displayId": "%s-LinearAlignmentsDisplay" % tId,
+                },
+            ],
+        }
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/jbrowse2/jbrowse2.py:949]: Maybe, it works so far for the tests that have been done by the VGP/ERGA community. I hope we can make this more stable as we go.

File: tools/jbrowse2/jb2_urlconf.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+inconf = open("config.json", "r").readlines()
+with open("config.json.local", "w") as bak:
+    bak.write("".join(inconf))
+urlbase = "https://galaxy.genomicsvl-students.cloud.edu.au/jbrowse/hum/"
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/jbrowse2/jb2_urlconf.py]: That file was not necessary, I removed it. Thanks for the review.

File: tools/jbrowse2/jbrowse2.py [hunk 4]
```diff
@@ -0,0 +1,1753 @@
...
+    def getNrow(self, url):
+        useuri = url.startswith("https://") or url.startswith("http://")
+        if not useuri:
+            fl = open(url, "r").readlines()
+            nrow = len(fl)
+        else:
+            try:
+                scontext = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+                scontext.check_hostname = False
+                scontext.verify_mode = ssl.VerifyMode.CERT_NONE
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/jbrowse2/jbrowse2.py:496]: Its a fallback. We had some users putting in URLs with invalid certs. By default, the cert is checked. If you prefer I can remove the fallback.

**Comment by bgruening** [comment]
I just updated to the latest version of JBrowse2, no issues so far, tests are passing :tada: 

@hexylena the repeat you requested in https://github.com/galaxyproject/tools-iuc/pull/5697#issuecomment-1911807744 is also integrated as far as I understood.

**Pull request closed without merging**